src/main/scala/com/fulcrumgenomics/bam/api/SamRecord.scala

```diff
@@ -185,6 +190,12 @@ trait SamRecord {
       mateStart + cigar.lengthOnTarget - 1 + cigar.reverseIterator.takeWhile(_.operator.isClipping).map(_.length).sum
     }
   }
+  @inline final def matesOverlap: Option[Boolean] = {
+    require(mapped && paired && mateMapped, "Cannot determine if mates overlap without paired mates that are both mapped.")
+    if (refIndex != mateRefIndex) Some(false)
+    else if (mateStart >= start && mateStart <= end) Some(true)
```

Comment:
if `end < mateStart` then I think we should return `Some(false)` right?

Reply:
I think you are right, we could code another fast path.

Reply:
I added the case and luckily I already have a unit test in place to trigger the branch.

I'll confirm all branches are actually tested once codecov tells me.